File: Season 1/07th_Competition/Jongryul/08/1021.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const arr = input[1].split(" ").map(Number);
const q = Array.from(Array(n), (_, i) => i + 1);
let cnt = 0;
for (let i = 0; i < m; i++) {
  if (arr[i] !== q[0]) {
    const idx = q.findIndex((e) => e === arr[i]);
    const gap = q.length - idx;
    if (idx > gap) {
      while (arr[i] !== q[0]) {
        q.unshift(q.pop());
        cnt++;
      }
    } else {
      while (arr[i] !== q[0]) {
        q.push(q.shift());
        cnt++;
      }
    }
  }
  q.shift();
}

console.log(cnt);

File: 01_Season/07th_Competition/Jongryul/08/1021.py
import sys
from collections import deque
input = sys.stdin.readline

n, m = map(int, input().split())
arr = list(map(int, input().split()))

q = deque([i for i in range(1, n+1)])
cnt = 0
for i in arr:
    if i == q[0]:
        q.popleft()
    else:
        idx = q.index(i)
        gap = len(q)-idx
        if idx > gap:
            q.rotate(gap)
            cnt += gap
        else:
            cnt += idx
            q.rotate(-idx)
        q.popleft()
print(cnt)